subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -1164,13 +1164,20 @@ uint8_t ll_df_set_conn_cte_req_enable(uint16_t handle, uint8_t enable,
 
 	if (!enable) {
 		ull_cp_cte_req_set_disable(conn);
+		/* This function is always executed from thread context. Its execution may be
+		 * interrupted by myflay job, executed as interrupt, that handles CTE_RSP.
+		 * This is why a memory barrier is used to make sure the CTE_REQ disable changes
+		 * reflected in memory.
+		 */
+		cpu_dmb();
```

Comment:
C Compiler on ARM may prioritize Memory Read before Memory Write and in the code construct here `cpu_dmb()` ensure that any memory write operation is complete before the memory read (`conn->llcp.cte_req.is_active`) happens.